src/mainwindow.hpp

```diff
@@ -248,5 +258,9 @@ class MainWindow : public QMainWindow
     QString compileCommand() const;
     int timeLimit() const;
     void updateCompileAndRunButtons() const;
+    void setStopwatch();
+
+    virtual void hideEvent(QHideEvent *event) override;
+    virtual void showEvent(QShowEvent *event) override;
```

Comment:
`onTabChanged` signal will only give the index of the currently changed tab. But we need to stop the stopwatch of the previous tab, before switching to a new tab. I was not able to find a way to achieve this only using `onTabChanged` signal.

Reply:
I believe the objective is: current visible tab should have active timer and others should have paused timer.

You could achieve it in 3 ways:

1. Add a member function to keep track of last tab, I believe in this case when restoring many tabs  on startup it will become more complex. 

2. Have a single Stopwatch attached to app window with multiple states for each tabs. 

3. On every tab switch iterate all other tabs and stop the timer if running except the current tab. Simplest but could be performance intensive (needs benchmark against current implementation).

Reply:
I think that the current implementation is simple that all 3 ways you suggested. Is there any reason for not preferring this solution?